### PR TITLE
Ds 6157 url issue with modisa l3b chl ms

### PR DESCRIPTION
**Pull request by mylesstokowski**
## Description

This PR is based on #41 and contains:
- a refactor to use Pydantic models to fix the broken URLs (hopefully more concise)
- tests proving that the fix works

Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] `pytest` should greenlight. You could read through the tests and/or poke around with a debugger
- [ ] @cearlefraym's testing script should still work, i.e. `python test.py`. **NOTE**  that we should remove this and any other extra testing files before merging to main.

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

## Next Steps
- [ ] Assign a reviewer based on the [code owner document](https://github.com/fraymio/modis_tools/blob/main/.github/CODEOWNERS).

- [ ] Once your review is approved, merge and delete the feature branch

On behalf of the Modis Tools Dev Team, thank you for your hard work! ✨

### CODE REVIEW

File: modis_tools/resources.py
Comment (cearlefraym) [inline modis_tools/resources.py:28]: this was originally:

```suggestion
            collection_feed = CollectionFeed(**resp.json()["feed"])
```

File: modis_tools/resources.py
Comment (cearlefraym) [inline modis_tools/resources.py:120]: Same as above 

```suggestion
                granule_feed = GranuleFeed(**resp.json()["feed"])
```

**Review by cearlefraym — APPROVED** [review]
So much cleaner. I came across this bit of documentation while looking into the `pre` arg: https://docs.pydantic.dev/1.10/usage/validators/#validate-always
Should we also include the `always` arg? All test are passing on my end, lgtm.

**Comment by mylesstokowski** [comment]
> Should we also include the always arg? 

Interesting, I wasn't familiar. I tried adding that parameter, and didn't observe any chance in behavior. I tested by running `test_space_is_removed_from_link()` as is and with the `href` field removed in the input.  Since it's working now, I'm tempted to leave it as is.

**Comment by mylesstokowski** [comment]
Thanks for the review @cearlefraym! One q: any preference whether I merge this into your base branch, then merge that to main, or alternatively switch the base branch to `main` and merge it directly? 

And relatedly, confirming there's no special approval process for this repo (i.e., confirming that we don't need another approval to merge to main)?

**Review by cearlefraym — APPROVED** [review]
Thanks for looking into that behavior @mylesstokowski, lgtm.
If this was started from my branch, we can go directly to main to avoid cleaning up the test files. 
For merging I think we just require one approval. I can approve once this is pointed to main.

File: modis_tools/models.py
Comment (mylesstokowski) [inline modis_tools/models.py:102]: I observed that it (`test.py`) broke when this was un-commented, but making the field optional worked. Proposal to do this b/c it's backwards compatible if there are cases where the `granule_size` field is provided by the API and then used downstream
```suggestion
    granule_size: Optional[float] = None
```

File: modis_tools/granule_handler.py
Comment (mylesstokowski) [inline modis_tools/granule_handler.py:192]: I stepped through this with a debugger – makes sense to me

**Review by cearlefraym — APPROVED** [review]
Looks great, clear to merge

**Pull request merged**
